syntax = "proto3";

package alyncoin;

message TransactionProto {
    string sender = 1;
    string recipient = 2;
    double amount = 3;
    uint64 timestamp = 4;

    bytes signature_dilithium = 5;
    bytes signature_falcon = 6;
    bytes zkproof = 7;

    bytes sender_pubkey_dilithium = 8;
    bytes sender_pubkey_falcon = 9;
    string metadata = 10;
    string hash = 11;

    bool is_system = 12;

    string amount_str = 13;
}
